EstimateMedianVal returns higher fee for higher confTarget

Running on recent master, commit f1dbf92ff0475a01d20170ea422c1d086acbbc57

If I run:
```
test/functional/feature_fee_estimation.py --randomseed 108574360997204915
```

I get:
```
2020-12-19T20:14:06.175000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_bcwexp78
2020-12-19T20:14:07.738000Z TestFramework (INFO): This test is time consuming, please be patient
2020-12-19T20:14:07.738000Z TestFramework (INFO): Splitting inputs so we can generate tx's
2020-12-19T20:14:10.870000Z TestFramework (INFO): Finished splitting
2020-12-19T20:14:12.104000Z TestFramework (INFO): Will output estimates for 1/2/3/6/15/25 blocks
2020-12-19T20:14:12.104000Z TestFramework (INFO): Creating transactions and mining them with a block size that can't keep up
2020-12-19T20:14:15.879000Z TestFramework (INFO): Creating transactions and mining them at a block size that is just big enough
2020-12-19T20:14:21.104000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/sdaftuar/ccl-bitcoin/2020-12-latest-master/test/functional/test_framework/test_framework.py", line 126, in main
    self.run_test()
  File "test/functional/feature_fee_estimation.py", line 256, in run_test
    check_estimates(self.nodes[1], self.fees_per_kb)
  File "test/functional/feature_fee_estimation.py", line 140, in check_estimates
    check_smart_estimates(node, fees_seen)
  File "test/functional/feature_fee_estimation.py", line 130, in check_smart_estimates
    % (feerate, last_feerate))
AssertionError: Estimated fee (0.000565) larger than last fee (0.000479) for lower number of confirms

```

Unfortunately this larger than life^H^H^Hast fee issue in `feature_fee_estimation.py` keeps popping up.

From https://cirrus-ci.com/task/4756253495787520:

```
AssertionError: Estimated fee (0.000400) larger than last fee (0.000338) for lower number of confirms
```

--- a/src/policy/fees.cpp
+++ b/src/policy/fees.cpp
@@ -259,6 +259,11 @@ double TxConfirmStats::EstimateMedianVal(int confTarget, double sufficientTxVal,
     unsigned int curFarBucket = maxbucketindex;
     unsigned int bestFarBucket = maxbucketindex;
 
+    // We'll always group buckets into sets that meet sufficientTxVal --
+    // this ensures that we're using consistent groups between different
+    // confirmation targets.
+    double partialNum = 0;
+
     bool foundAnswer = false;
     unsigned int bins = unconfTxs.size();
     bool newBucketRange = true;
@@ -274,6 +279,7 @@ double TxConfirmStats::EstimateMedianVal(int confTarget, double sufficientTxVal,
         }
         curFarBucket = bucket;
         nConf += confAvg[periodTarget - 1][bucket];
+        partialNum += txCtAvg[bucket];
         totalNum += txCtAvg[bucket];
         failNum += failAvg[periodTarget - 1][bucket];
         for (unsigned int confct = confTarget; confct < GetMaxConfirms(); confct++)
@@ -283,7 +289,14 @@ double TxConfirmStats::EstimateMedianVal(int confTarget, double sufficientTxVal,
         // we can test for success
         // (Only count the confirmed data points, so that each confirmation count
         // will be looking at the same amount of data and same bucket breaks)
-        if (totalNum >= sufficientTxVal / (1 - decay)) {
+
+        if (partialNum < sufficientTxVal / (1 - decay)) {
+            // the buckets we've added in this round aren't sufficient
+            // so keep adding
+            continue;
+        } else {
+            partialNum = 0; // reset for the next range we'll add
+
             double curPct = nConf / (totalNum + failNum + extraNum);
 
             // Check to see if we are no longer getting confirmed at the success rate

